### PR TITLE
feat(packages): fix reserve detail price derivation for multi-reserve deployments

### PR DESCRIPTION
**Pull request by jonybur**

### CODE REVIEW

**Review by jrwbabylonlab — DISMISSED** [review]

File: services/vault/src/applications/aave/components/Detail/hooks/useAaveReserveDetail.ts [hunk 1]
```diff
@@ -118,30 +118,33 @@ export function useAaveReserveDetail({
...
   // `services/vault/src/clients/eth-contract/chainlink/query.ts` once feeds
   // are available for borrowable reserves on our target network. That would
-  // remove the single-reserve assumption and the stablecoin fallback.
+  // remove the single-debt-reserve assumption and the stablecoin fallback.
   const tokenPriceUsd = useMemo((): number | null => {
     if (!selectedReserve) return null;
 
     if (currentDebtAmount > 0 && debtValueUsd > 0) {
-      if (borrowableReserves.length !== 1) {
+      const userDebtReserveCount = position?.debtPositions?.size ?? 0;
```
Comment (gbarkhatov) [inline services/vault/src/applications/aave/components/Detail/hooks/useAaveReserveDetail.ts]: Nit: **Edge case — `debtPositions` undefined vs empty Map**: When `debtPositions` is `undefined` (no borrowable reserves provided to the position fetcher), `size` would be `undefined`, and `?? 0` yields `0`. Since `currentDebtAmount > 0` already guards this path, the code is safe. However, the semantic meaning differs: *we don't know the user's debt positions* vs *the user has no debt positions*. Currently both result in the same behavior (fall through to stablecoin fallback), which is fine.

**Review by gbarkhatov — DISMISSED** [review]
Nit: **No test coverage** — There are no unit tests for `useAaveReserveDetail`. This is pre-existing, not introduced by this PR, but given that this hook handles financial price derivation with multiple edge cases (single debt reserve, multi-debt reserve, stablecoin fallback, unknown token), it would benefit from tests. Not a blocker for this PR.
